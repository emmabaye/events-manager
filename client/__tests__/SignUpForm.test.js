import { BrowserRouter } from 'react-router-dom';
import ConnectedSignUpForm, { SignUpForm } from '../components/SignUpForm.jsx';

describe('SignUpForm Component', () => {
  const initialState = {
    authReducer: {
      status: "Error",
      message: "",
    }
  };

  const store = mockStore(initialState);

  const props = {
    dispatch: () => {}
  };

  global.localStorage = {
    getItem: (str) => str
  };

  describe('SignUp page  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = mount(<BrowserRouter><SignUpForm /></BrowserRouter>);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render for connected component', () => {
      const wrapper = mount(<BrowserRouter><Provider store={store}><ConnectedSignUpForm /></Provider></BrowserRouter>);
      expect(wrapper.length).toEqual(1);
    });

    it('it should render redirect on  success', () => {
      const wrapper = shallow(<BrowserRouter><SignUpForm {...props} status="Success" /></BrowserRouter>);
      expect(wrapper.length).toEqual(1);
    });
  });

  describe('SignUp page events', () => {
  
    it('it should handle onChange event', () => {
      const wrapper = shallow(<SignUpForm {...props} />);
      const emailInput = wrapper.find('[name="email"]');
      emailInput.simulate('change', { target: {} });
    });

    it('it should submit form', () => {
      const wrapper = mount(<BrowserRouter><Provider store={store}><ConnectedSignUpForm /></Provider></BrowserRouter>);
      const button = wrapper.find('.btn');
      button.simulate('click', { preventDefault: () => {} });
    });
  });
});
