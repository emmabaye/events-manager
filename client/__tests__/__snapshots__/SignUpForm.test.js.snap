// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm Component SignUp page  should render self it should render for dumb component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SignUpForm>
      <div>
        <NavBar
          page="SignUpForm"
        >
          <nav
            className="navbar navbar-toggleable-md navbar-light bg-faded"
          >
            <button
              aria-controls="navbarText"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler navbar-toggler-right"
              data-target="#navbarText"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
            <Link
              className="navbar-brand"
              replace={false}
              to="/"
            >
              <a
                className="navbar-brand"
                href="/"
                onClick={[Function]}
              >
                EVENTS MANAGER
              </a>
            </Link>
            <div
              className="collapse navbar-collapse"
              id="navbarText"
            >
              <ul
                className="navbar-nav ml-auto"
              >
                <li
                  className="nav-item active"
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to="/"
                  >
                    <a
                      className="nav-link"
                      href="/"
                      onClick={[Function]}
                    >
                      Home 
                      <span
                        className="sr-only"
                      >
                        (current)
                      </span>
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to="/centers"
                  >
                    <a
                      className="nav-link"
                      href="/centers"
                      onClick={[Function]}
                    >
                      Centers
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to="/login"
                  >
                    <a
                      className="nav-link"
                      href="/login"
                      onClick={[Function]}
                    >
                      Sign In
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to="/signup"
                  >
                    <a
                      className="nav-link"
                      href="/signup"
                      onClick={[Function]}
                    >
                      Sign Up
                    </a>
                  </Link>
                </li>
              </ul>
            </div>
          </nav>
        </NavBar>
        <div
          className="container signup "
        >
          <div
            className="row"
          >
            <div
              className="container"
            >
              <form
                action="true"
              >
                <div
                  className="form-group row"
                >
                  <label
                    className="col-sm-3 col-form-label"
                  />
                  <div
                    className="col-sm-9"
                  />
                </div>
                <div
                  className="form-group row"
                >
                  <label
                    className="col-sm-3 col-form-label"
                    htmlFor="inputEmail3"
                  >
                    First Name
                  </label>
                  <div
                    className="col-sm-9"
                  >
                    <input
                      className="form-control"
                      name="firstName"
                      onChange={[Function]}
                      placeholder="First Name"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="form-group row"
                >
                  <label
                    className="col-sm-3 col-form-label"
                    htmlFor="inputEmail3"
                  >
                    Last Name
                  </label>
                  <div
                    className="col-sm-9"
                  >
                    <input
                      className="form-control"
                      name="lastName"
                      onChange={[Function]}
                      placeholder="Last Name"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="form-group row"
                >
                  <label
                    className="col-sm-3 col-form-label"
                    htmlFor="inputEmail3"
                  >
                    Email
                  </label>
                  <div
                    className="col-sm-9"
                  >
                    <input
                      className="form-control"
                      name="email"
                      onChange={[Function]}
                      placeholder="Email"
                      type="email"
                    />
                  </div>
                </div>
                <div
                  className="form-group row"
                >
                  <label
                    className="col-sm-3 col-form-label"
                    htmlFor="inputPassword3"
                  >
                    Password
                  </label>
                  <div
                    className="col-sm-9"
                  >
                    <input
                      className="form-control"
                      name="password"
                      onChange={[Function]}
                      placeholder="Password"
                      type="password"
                    />
                  </div>
                </div>
                <div
                  className="form-group row"
                >
                  <div
                    className="offset-sm-3 col-sm-9"
                  >
                    <button
                      className="btn btn-primary"
                      onClick={[Function]}
                      type="submit"
                    >
                      Sign up
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <Footer>
          <footer
            className="container-fluid"
          >
            <p
              className="h6 text-center"
            >
              <i
                className="fa fa-copyright"
              />
               2017 EVENTS MANAGER
            </p>
          </footer>
        </Footer>
      </div>
    </SignUpForm>
  </Router>
</BrowserRouter>
`;
